### PR TITLE
TrungP - String conversion

### PR DESCRIPTION
**Pull request by crsici**

### CODE REVIEW

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Date;
+
+/**
+ * Created by sic.org on 8/2/2016.
+ */
+public class JavaStringConversionUnitTest {
+    @Test
+    public void stringToboolean() {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: the name does not follow the naming convention. You can say stringToPrimitiveBoolean. Follow similar norm in other test methods as well.

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    @Test
+    public void stringToFloat() {
+        Float stringToFloat = Float.valueOf("1.1");
+
+        System.out.println(stringToFloat);
+    }
+
+    @Test
+    public void stringTodouble() {
+        double stringToDouble = Double.parseDouble("1");
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: May be we can use some double values in here.

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    @Test
+    public void whenConvertingStringToPrimitiveCharArray_thenCorrect() {
+        String primitiveChars = "Hello world";
+        char[] chars = primitiveChars.toCharArray();
+
+        assertThat(Arrays.equals(chars,new char[]{'H','e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd'})).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveByteArray_default_thenCorrect() {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: The test name is not following **givenX_whenY_thenZ** pattern. Maybe we can say **whenConvertingStringToPrimitiveByteArrayWithDefaultCharset_thenCorrect**

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        String primitiveBytes = "Hello world";
+        byte[] bytes = primitiveBytes.getBytes();
+
+        System.out.println(Arrays.toString(bytes));
+    }
+
+    @Test
+    public void stringToPrimitiveByteArray_CharsetName_thenCorrect() throws UnsupportedEncodingException {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: **whenConvertingStringToPrimitiveByteArrayWithCharsetName_thenCorrect**

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        String primitiveBytes = "Hello world";
+        byte[] bytes = primitiveBytes.getBytes("UTF-8");
+
+        System.out.println(Arrays.toString(bytes));
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveByteArray_Charset_thenCorrect() {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: **whenConvertingStringToPrimitiveByteArrayWithCharset_thenCorrect**

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java (hunk 3 above)
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: The test name is not following **givenX_whenY_thenZ** naming convention. May be we can say - 
**givenString_whenConvertingToPrimitiveByteArrayUsingDefaultCharset_thenCorrect**? What do you think?

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java (hunk 4 above)
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: Similarly, we can say - **givenString_whenConvertingToByteArrayWithCharsetName_thenCorrect**

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java (hunk 5 above)
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: **givenString_whenConvertingToByteArrayWithCharset_thenCorrect**

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        String primitiveBytes = "Hello world";
+        byte[] bytes = primitiveBytes.getBytes(StandardCharsets.UTF_8);
+
+        System.out.println(Arrays.toString(bytes));
+    }
+
+    @Test
+    public void whenConvertingStringToDate_java7_thenCorrect() throws ParseException {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: **givenString_whenConvertingToDateJava7_thenCorrect**

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+        SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy");
+        String dateInString = "26/06/2016";
+        Date date = formatter.parse(dateInString);
+
+        assertThat(date.equals(Date.from(LocalDate.of(2016,06,26).atStartOfDay(ZoneId.systemDefault()).toInstant()))).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToDate_java8_thenCorrect() {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: givenString_whenConvertingToDateJava8_thenCorrect

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+        LocalDate localDate = LocalDate.parse("2016-06-26", DateTimeFormatter.ISO_LOCAL_DATE);
+
+        assertThat(localDate.equals(LocalDate.of(2016,06,26))).isTrue();
+    }
+
+    @Test
+    public void whenConvertingDateToString_java7_thenCorrect() {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: givenDate_whenConvertingToStringJava7_thenCorrect

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+        SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy");
+        String dateToString = formatter.format(Date.from(LocalDate.of(2016,06,26).atStartOfDay(ZoneId.systemDefault()).toInstant()));
+
+        assertThat(dateToString.equals("26/06/2016")).isTrue();
+    }
+
+    @Test
+    public void whenConvertingDateToString_java8_thenCorrect() {
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: givenDate_whenConvertingToStringJava8_thenCorrect

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+        Boolean wrapperBoolean = Boolean.valueOf("false");
+
+        Assert.assertFalse(wrapperBoolean);
+    }
+
+    @Test
+    public void givenString_whenConvertingToPrimitiveByte_thenCorrect() {
+        byte primitiveByte = Byte.parseByte("1");
+
+        assertThat(primitiveByte == 1).isTrue();
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: assertThat(primitiveByte).isEqualTo(1);

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    @Test
+    public void givenString_whenConvertingToWrapperByte_thenCorrect() {
+        Byte wrapperByte = Byte.valueOf("1");
+
+        assertThat(wrapperByte == 1).isTrue();
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: It is better to follow **assertThat(primitiveByte).isEqualTo(1)** pattern when comparing expected value with actual value for different types.

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+
+        assertThat(Arrays.equals(chars,new char[]{'H','e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd'})).isTrue();
+    }
+
+    @Test
+    public void givenString_whenConvertingToPrimitiveByteArray_default_thenCorrect() {
+        String primitiveBytes = "Hello world";
+        byte[] bytes = primitiveBytes.getBytes();
+
+        System.out.println(Arrays.toString(bytes));
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: missing assert?

File: core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    @Test
+    public void givenString_whenConvertingToPrimitiveByteArray_CharsetName_thenCorrect() throws UnsupportedEncodingException {
+        String primitiveBytes = "Hello world";
+        byte[] bytes = primitiveBytes.getBytes("UTF-8");
+
+        System.out.println(Arrays.toString(bytes));
```
Comment (pkhanal) [inline core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java]: missing assert?

**Comment by pivovarit** [comment]
@pkhanal @crsici Is this one still "alive"? If not, please close it

**Comment by pivovarit** [comment]
This one looks dead to me. If not, let me know

**Pull request closed without merging**